drivers/sensor/tdk/icm45686/icm45686.c

```diff
@@ -0,0 +1,363 @@
+				 enum sensor_channel chan)
+{
+	if (chan != SENSOR_CHAN_ALL) {
+		return -ENOTSUP;
+	}
+
+	struct icm45686_data *data = dev->data;
+
+	return icm45686_fetch_data(dev, &data->edata);
+}
+
+static int icm45686_channel_get(const struct device *dev,
+				enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct icm45686_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.x,
+				  &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.y,
+				  &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.z,
+				  &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.x,
+				   &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.y,
+				   &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.z,
+				   &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_DIE_TEMP:
+		icm45686_temp_c(data->edata.payload.temp, &val->val1, &val->val2);
+		break;
+	case SENSOR_CHAN_ACCEL_XYZ:
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.x,
+				  &val[0].val1, &val[0].val2);
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.y,
+				  &val[1].val1, &val[1].val2);
+		icm45686_accel_ms(&data->edata, data->edata.payload.accel.z,
+				  &val[2].val1, &val[2].val2);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.x,
+				   &val->val1, &val->val2);
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.y,
+				   &val[1].val1, &val[1].val2);
+		icm45686_gyro_rads(&data->edata, data->edata.payload.gyro.z,
+				   &val[2].val1, &val[2].val2);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#if defined(CONFIG_SENSOR_ASYNC_API)
+
+static inline void icm45686_submit_one_shot(const struct device *dev,
+					    struct rtio_iodev_sqe *iodev_sqe)
+{
+	const struct sensor_read_config *cfg = iodev_sqe->sqe.iodev->data;
+	const struct sensor_chan_spec *const channels = cfg->channels;
+	const size_t num_channels = cfg->count;
+	uint32_t min_buf_len = sizeof(struct icm45686_encoded_data);
+	int err;
+	uint8_t *buf;
+	uint32_t buf_len;
+	struct icm45686_encoded_data *edata;
+
+	err = rtio_sqe_rx_buf(iodev_sqe, min_buf_len, min_buf_len, &buf, &buf_len);
+	if (err != 0) {
+		LOG_ERR("Failed to get a read buffer of size %u bytes", min_buf_len);
+		rtio_iodev_sqe_err(iodev_sqe, err);
+		return;
+	}
+
+	edata = (struct icm45686_encoded_data *)buf;
+
+	err = icm45686_encode(dev, channels, num_channels, buf);
+	if (err != 0) {
+		LOG_ERR("Failed to encode sensor data");
+		rtio_iodev_sqe_err(iodev_sqe, err);
+		return;
+	}
+
+	err = icm45686_fetch_data(dev, edata);
```

Comment:
Same story as the pixart driver, could easily convert this to use an async read and avoid needing RTIO_WORKQ which, imho, is a good thing to try for though not a blocker

Reply:
Sounds good. I've removed the RTIO Workqueue dependency